API/projectClass/projectClass.m

```diff
@@ -229,12 +240,47 @@ function delete(obj)
         end
         
         function obj = addLayer(obj, varargin)
-            % Adds a single layer to the layers object. Expects layer
-            % details which are name, thickness, SLD, roughness,
-            % hydration, and hydrate with, or provide with no details,
-            % or just a name, to create an empty layer.
+            % Adds a new layer to the project. This method can be called in 2 ways. The first for when ``absorption`` is false  
+            %
+            % ``addLayer(name, thickness, realSLD, roughness, hydration, hydrateWith)``
+            % 
+            % and the second includes an extra argument imaginarySLD for when ``absorption`` is true.
+            % 
+            % ``addLayer(name, thickness, realSLD, imaginarySLD, roughness, hydration, hydrateWith)``
             %
-            % project.addLayer('New Layer');
+            % Examples
+            % --------
+            % To add a new layer with name only.
+            % 
+            % >>> project.addLayer(, 'New layer');
+            % 
+            % To add a new layer when ``absorption`` is false.
+            % 
+            % >>> project.addLayer('Water Layer', 'Water thickness', 'Water SLD', 'Bilayer heads roughness', 'Water hydration', 'Bulk out');
+            % 
+            % To add a new layer when ``absorption`` is true.
+            % 
+            % >>> project.addLayer('Layer 1', 'Layer thickness', 'Layer Real SLD', 'Layer Imaginary SLD', 'Layers roughness, 'Layer hydration', 'Bulk in');
+            %  
+            % Parameters
+            % ----------
+            % name : string or char array, default: auto-generated name
+            %     The name of this layer.
+            % thickness : string or char array or whole number, default: ''
+            %     The name (or the row index) of the parameter describing the thickness of this layer.
+            % realSLD : string or char array or whole number, default: ''
+            %     The name (or the row index) of the parameter describing the real (scattering) term
+            %     for the scattering length density of this layer.
+            % imaginarySLD : string or char array or whole number, default: ''
+            %     The name (or the row index) of the parameter describing the imaginary (``absorption``) term
```

Comment:
As before, be consistent with (scattering) and (``absorption``)

Reply:
Will fix, I lifted from python-RAT https://github.com/RascalSoftware/python-RAT/blob/b92118a70748ceef790139ed7dbe040933c15afb/RATapi/models.py#L504

Reply:
Hold on, I'd say the actual words "scattering" and "absorption" are correct, it's the punctuation around then that I wanted to change. The python has them as (scattering) and (absorption) and I'd suggest copying that.